lib/Arch/Sleigh/ControlFlowStructuring.cpp

```diff
@@ -410,31 +428,67 @@ ControlFlowStructureAnalysis::ComputeCategory(
     DLOG(ERROR) << "No coarse category found";
     return std::nullopt;
   }
-  auto context_updater = this->BuildContextUpdater(std::move(entry_context));
+  auto context_updater = this->BuildContextUpdater(entry_context);
+
+  if (*maybe_ccategory == CoarseCategory::kCatNormalWithIntraInstructionFlow) {
+    // our control flow analysis for decoding contexts doesnt handle intraprocedural control flow,
+    // so if we have a normal instruction with no decoding context updates then we are fine otherwise bail
+    auto no_context_updates = std::all_of(
+        ops.begin(), ops.end(), [&context_updater](const RemillPcodeOp &op) {
+          return !op.outvar.has_value() ||
+                 !context_updater.GetRemillReg(*op.outvar).has_value();
+        });
+
+    if (no_context_updates) {
+      Instruction::NormalInsn norm_insn(
+          (Instruction::FallthroughFlow(entry_context)));
+      Instruction::InstructionFlowCategory ifc = norm_insn;
+      return std::make_pair(ifc, std::nullopt);
+    } else {
+      DLOG(ERROR)
+          << "Had an instructon with intrainstruction flow, but also decoding context updates";
+      return std::nullopt;
+    }
+  }
+
   auto flows = GetBoundContextsForFlows(ops, cc, context_updater);
 
   switch (*maybe_ccategory) {
+    case CoarseCategory::kCatNormalWithIntraInstructionFlow:
+      return std::nullopt;
     case CoarseCategory::kCatAbnormal: return ExtractAbnormal(flows, ops);
     case CoarseCategory::kCatConditionalAbnormal:
       return ExtractConditionalAbnormal(flows, ops);
     case CoarseCategory::kCatNormal: return ExtractNormal(flows, ops);
   }
 }
 
+std::optional<std::string>
+ContextUpdater::GetRemillReg(const VarnodeData &outvar) {
+  auto reg_name =
+      this->engine.getRegisterName(outvar.space, outvar.offset, outvar.size);
+  auto maybe_remill_reg_name = this->context_reg_mapping.find(reg_name);
+  if (maybe_remill_reg_name != this->context_reg_mapping.end()) {
+    return maybe_remill_reg_name->second;
+  } else {
```

Comment:
Drop `else` and all those `this->`.

Reply:
I would also change `maybe_remill_reg_name` to simply `it` (since technically it is an iterator and function is short enough to not really suffer from loss of what is being retrieved).

Reply:
[ddcbe39](https://github.com/lifting-bits/remill/pull/645/commits/ddcbe397f88ca11fbc6d2e93acf240f4e5611a2d)